src/main/java/org/prebid/server/spring/config/AnalyticsConfiguration.java

```diff
@@ -52,6 +57,54 @@ LogAnalyticsReporter logAnalyticsReporter(JacksonMapper mapper) {
         return new LogAnalyticsReporter(mapper);
     }
 
+    @Configuration
+    @ConditionalOnProperty(prefix = "analytics.greenbids", name = "enabled", havingValue = "true")
+    public static class GreenbidsAnalyticsConfiguration {
+
+        @Bean
+        GreenbidsAnalyticsReporter greenbidsAnalyticsReporter(
+                GreenbidsAnalyticsConfigurationProperties greenbidsAnalyticsConfigurationProperties,
+                GreenbidsJacksonMapper jacksonMapper,
+                HttpClient httpClient,
+                Clock clock,
+                PrebidVersionProvider prebidVersionProvider) {
+            return new GreenbidsAnalyticsReporter(
+                    greenbidsAnalyticsConfigurationProperties.toComponentProperties(),
+                    jacksonMapper,
+                    httpClient,
+                    clock,
+                    prebidVersionProvider);
+        }
```

Comment:
your jackson mapper is not defined as a spring bean, so the config won't work, please try to run the app to double-check

Reply:
Yes you are right, after running App the bean `GreenbidsJacksonMapper` is not found. To fix it I've added beans into `AnalyticsConfiguration`

```java
@Bean
public ObjectMapper greenbidsObjectMapper() {
    return new ObjectMapper();
}

@Bean
public GreenbidsJacksonMapper greenbidsJacksonMapper(ObjectMapper greenbidsObjectMapper) {
    final ObjectMapper greenbidsCustomizedMapper = greenbidsObjectMapper
            .setPropertyNamingStrategy(PropertyNamingStrategy.LOWER_CAMEL_CASE);
    return new GreenbidsJacksonMapper(greenbidsCustomizedMapper);
}

```

Reply:
After that have rerun the App which worked

```json
{"type": "greenbids/publishers/exceptions/TestRequest", "title": "TestRequest", "status": 202, "detail": "Request sent with the test pbuid"}
{"version":"2.2.0","auctionId":"1","referrer":"http://example.com/prebid_server_test.html","sampling":1.0,"prebidServer":"pbs-java/3.0.0","greenbidsId":"26180c31-5f3a-4417-942c-283be18e2302","pbuid":"PBUID_FROM_GREENBIDS","billingId":"3ef30740-ffb1-4acd-8c7b-e3fa537b47e6","adUnits":[{"code":"a","unifiedCode":{"value":"a","source":"adUnitCodeSource"},"mediaTypes":{"banner":{"sizes":[[320,50]],"pos":null,"name":null},"video":null,"native":null},"bids":[{"bidder":"rubicon","isTimeout":false,"hasBid":true},{"bidder":"rubicon","isTimeout":false,"hasBid":false}]}],"auctionElapsed":0}

```